### PR TITLE
fs/userfs: destroy nxmutex properly

### PR DESCRIPTION
**Pull request by anchao**
## Summary

fs/userfs: destroy nxmutex properly

Signed-off-by: chao an <anchao@xiaomi.com>

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

File: drivers/misc/rwbuffer.c [hunk 1]
```diff
@@ -842,6 +843,19 @@ int rwb_initialize(FAR struct rwbuffer_s *rwb)
...
           ferr("Read-ahead buffer kmm_malloc(%" PRIu32 ") failed\n",
           allocsize);
+          nxmutex_destroy(&rwb->rhlock);
+#ifdef CONFIG_DRVR_WRITEBUFFER
+          if (rwb->wrmaxblocks > 0)
+            {
+              nxmutex_destroy(&rwb->wrlock);
+            }
+
+          if (rwb->wrbuffer)
```
Comment (pkarashchenko) [inline drivers/misc/rwbuffer.c:853]: Optional

```suggestion
          if (rwb->wrbuffer != NULL)
```

Comment (anchao) [reply drivers/misc/rwbuffer.c:853]: Done

File: drivers/clk/clk_rpmsg.c [hunk 2]
```diff
@@ -485,18 +485,17 @@ static int64_t clk_rpmsg_sendrecv(FAR struct rpmsg_endpoint *ept,
...
-      return ret;
-    }
-
-  ret = nxsem_wait_uninterruptible(&cookie.sem);
-  if (ret < 0)
+  if (ret >= 0)
     {
-      return ret;
+      ret = nxsem_wait_uninterruptible(&cookie.sem);
+      if (ret == 0)
```
Comment (xiaoxiang781216) [inline drivers/clk/clk_rpmsg.c:491]:
```suggestion
      if (ret >= 0)
```

File: drivers/ioexpander/ioe_rpmsg.c [hunk 3]
```diff
@@ -275,18 +275,17 @@ static int ioe_rpmsg_sendrecv(FAR struct rpmsg_endpoint *ept,
...
-      return ret;
-    }
-
-  ret = rpmsg_wait(ept, &cookie.sem);
-  if (ret < 0)
+  if (ret >= 0)
     {
-      return ret;
+      ret = rpmsg_wait(ept, &cookie.sem);
+      if (ret == 0)
```
Comment (xiaoxiang781216) [inline drivers/ioexpander/ioe_rpmsg.c:281]:
```suggestion
      if (ret >= 0)
```

File: drivers/power/supply/regulator_rpmsg.c [hunk 4]
```diff
@@ -495,18 +495,17 @@ static int regulator_rpmsg_sendrecv(FAR struct rpmsg_endpoint *ept,
...
     {
-      return ret;
-    }
-
-  ret = nxsem_wait_uninterruptible(&cookie.sem);
-  if (ret < 0)
-    {
-      return ret;
+      ret = nxsem_wait_uninterruptible(&cookie.sem);
+      if (ret == 0)
```
Comment (xiaoxiang781216) [inline drivers/power/supply/regulator_rpmsg.c:501]:
```suggestion
      if (ret >= 0)
```

File: drivers/clk/clk_rpmsg.c (hunk 2 above)
Comment (anchao) [reply drivers/clk/clk_rpmsg.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**